pywbem_mock/_wbemconnection_mock.py

```diff
@@ -2913,9 +2917,21 @@ def _fake_invokemethod(self, methodname, objectname, Params, **params):
         namespace = localobject.namespace
         try:
             methodsrepo = self._get_methods_repo(namespace)
+        except CIMError as ce:
+            # If invoke indicates method repo not built, create it.
+            if ce.status_code == CIM_ERR_INVALID_NAMESPACE:
+                if namespace in self.classes or \
+                        namespace in self.instances:
+                    raise CIMError(CIM_ERR_NOT_SUPPORTED,
```

Comment:
Two issues: 

1. According to DSP0200, CIM_ERR_NOT_SUPPORTED in InvokeMethod means that the WBEM server generally does not support method invocations. That is not the case for this situation. DSP0200 offers CIM_ERR_METHOD_NOT_FOUND for the case where the class does not expose the method.
invocations.)
2. If we want to raise the error that the method does not exist on the class (which I understand the intention is), then this check for it is incorrect, IMO, and one of the checks further down is probably close.
3. The checks further down seem pretty convoluted, with error messages that have subtle nuances in what they say, and that don't tell clearly what's actually wrong.